server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentQuery.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.component;
+
+import java.util.Locale;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.db.WildcardPosition;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.sonar.db.MoreDatabaseUtils.buildLikeValue;
+
+public class ComponentQuery {
+  private final String nameOrKeyQuery;
+  private final String[] qualifiers;
+  private final String language;
+  private final Set<Long> componentIds;
+
+  /**
+   * Used by Dev Cockpit 1.9.
+   * Could be removed when Developer Cockpit doesn't use it anymore.
+   *
+   * @deprecated since 5.4, use {@link Builder} instead
+   */
+  @Deprecated
+  public ComponentQuery(@Nullable String nameOrKeyQuery, String... qualifiers) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)